### PR TITLE
Another validate_accounting script.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Attempt at a algod/indexer comparision script with fewer moving parts.

I got carried away here. The shell script worked pretty well but was slow so I ported it to go.

Both  implementations take the approach of accepting accounts from standard in, so a secondary program is needed to feed accounts to the script.

### CODE REVIEW

File: cmd/validator/dynamic_processor.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
+package main
```
Comment (winder) [inline cmd/validator/dynamic_processor.go:1]: This file has a comparison function that follows the same algorithm as the jq normalization in the bash script. The output it provides on a mismatch isn't as good as the other algorithm, so this is mainly for reference....

File: cmd/validator/struct_processor.go [hunk 2]
```diff
@@ -0,0 +1,466 @@
+package main
```
Comment (winder) [inline cmd/validator/struct_processor.go:1]: This is the main comparison function. It serialized the responses into a struct and compares them field by field. The report on a difference is pretty nice and will tell you exactly which fields are different.

File: cmd/validator/validate_accounting.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
+package main
```
Comment (winder) [inline cmd/validator/validate_accounting.go:1]: Main program, args, concurrency, reporting, etc

File: misc/generate_accounts.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
```
Comment (winder) [inline misc/generate_accounts.sh:1]: A better script is needed to stream accounts from postgres. Rotem gave me a python script that exports accounts directly from algod, so I exported them to a file. My current tests are using a file to feed the validator tool.

File: misc/validate_accounting.sh [hunk 5]
```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bash
```
Comment (winder) [inline misc/validate_accounting.sh:1]: For reference only. This script works fine but it's too slow to run against all mainnet accounts.

**Review by brianolson — COMMENTED** [review]
It's not a complete system, but if you find it useful then okay. A couple tweak suggestions.

**Pull request merged**